Model/ModelManager.php

```diff
@@ -322,20 +323,27 @@ public function getIdentifierValues($entity)
         //    throw new \RuntimeException('Entities passed to the choice field must be managed');
         //}
 
-
-        $class = $this->getMetadata(ClassUtils::getClass($entity));
+        $class = ClassUtils::getClass($entity);
+        $metadata = $this->getMetadata($class);
+        $platform = $this->getEntityManager($class)->getConnection()->getDatabasePlatform();
 
         $identifiers = array();
 
-        foreach ($class->getIdentifierValues($entity) as $value) {
+        foreach ($metadata->getIdentifierValues($entity) as $name => $value) {
             if (!is_object($value)) {
                 $identifiers[] = $value;
                 continue;
             }
+            $fieldType = $metadata->getTypeOfField($name);
+            $type = Type::getType($fieldType);
+            if ($type) {
+                $identifiers[] = $type->convertToDatabaseValue($value, $platform);
```

Comment:
@Soullivaneuh: This is needed to avoid:
>  RuntimeException: No entity manager defined for class Ramsey\Uuid\Uuid

when using child admin list of an admin whose entity has a custom identifier type.